modules/cas_cache/layer_cache_management.c

```diff
@@ -406,15 +425,17 @@ struct _cache_mngt_attach_context {
 };
 
 
-static void cache_start_rollback(struct work_struct *work)
+static int cache_start_rollback(void *data)
 {
 	struct cache_priv *cache_priv;
 	ocf_queue_t mngt_queue = NULL;
-	struct _cache_mngt_attach_context *ctx =
-		container_of(work, struct _cache_mngt_attach_context, work);
+	struct _cache_mngt_attach_context *ctx = data;
 	ocf_cache_t cache = ctx->cache;
 	int result;
 
+	if (kthread_should_stop())
```

Comment:
Shouldn't you put the module in this path as well?

Reply:
This condition is true only if synchronous `kthread_stop()` was called. In this case it is caller responsibility to put module.